### DISCUSSION TITLE
Default Collection Field Data Responses for API

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#4128] [Feature Requests]
Many users are not aware of the `fields` parameter or find it complex/cumbersome. However, this parameter is extremely important to the REST API since it defines how to fetch relational field data. Right now this param is only additive, meaning you build up what you want to be returned using parent field keys, relational field keys, or wildcards.

The problem is that many users expect to see their relational data in API responses... by default. Learning about `fields` solves this, but there might be other ways to make this more intuitive.

One option would be to allow configuring (via the App/API) the parent/relational fields that are returned by default when using `/items`. It's basically a way to set default values for the `fields` parameter per collection... either globally or per role.

I'm imagining a tree-view UI of all available parent/relational fields (deep fields load async), and you can toggle them on/off. This would likely need to be configured on the role-level to avoid complications of merging.

Field types could then have defaults for if they should be on by default... such as File/Files.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
